### PR TITLE
frontend/fix: #1043 added info when driver location is not available …

### PR DESCRIPTION
**Pull request by aryanguptaSG**
…in customer app

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
As a rider, I want the accurate location of the driver before pickup so that I am sure that the driver is moving towards my location . if driver is not connected to the internet we are showing a notification to customer app that driver is not connected to internet .

iOS PR for in app notification :- [link](https://bitbucket.juspay.net/projects/BEC/repos/atlas-ios/pull-requests/87/overview)
 
figma :- [link](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=58072%3A11094&t=OWYlVsorPJrLuOkL-1)
output android :- [link](https://drive.google.com/file/d/1G2HsA_rI0CijWHaVjORTcJiBLHAaaAy1/view?usp=sharing)
output iOS :-  [1](https://drive.google.com/file/d/1eem2Ty2JYfX6d_MdBKve9fIjByioVznt/view?usp=sharing) , [2](https://drive.google.com/file/d/17_FdwGAHwDIhbP5m0ts_x8nOEqs7J228/view?usp=sharing) , [3](https://drive.google.com/file/d/1RizR9LOIFfyGviCegob3ZE-xFjL8aCru/view?usp=sharing)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 1]
```diff
@@ -1863,6 +1864,11 @@ driverLocationTracking push action driverArrivedAction updateState duration trac
                                       specialLocationConfig destSpecialTagIcon sourceSpecialTagIcon
                                     else
                                       specialLocationConfig sourceSpecialTagIcon destSpecialTagIcon
+            let lastUpdate = getExpiryTime (resp^. _lastUpdate) true
+            if lastUpdate > 20 && state.props.currentStage == RideAccepted then do
+              _ <- liftFlow $ showInAppNotification (getString INTERNET_ISSUE_AT_DRIVERS_END) (getString YOU_MAY_CONTACT_DRIVER_TO_TRACK_YOUR_RIDE) "" "" "" "" "" "driver_location_error" 5000 "true" "false"
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1869]: @aryanguptaSG If it is a boolean then send boolean no need of string.

Comment (Vignesh-772) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1869]: Do the changes accordingly

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java [hunk 2]
```diff
@@ -121,7 +124,13 @@ private Notification(String channelId){
             counterView.setVisibility(View.GONE);
         }
 
-        private void setContent(String title , String message , String onTapEvent, String action1Text , String action2Text , String action1Image , String action2Image){
+        private void setContent(String title , String message , String onTapEvent, String action1Text , String action2Text , String action1Image , String action2Image , String showLoaderImage){
+            ProgressBar loaderImage = view.findViewById(R.id.loader);
+            if(showLoaderImage.equals("true")){
```
Comment (shanm-16) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java:129]: is this formatting correct?

File: Frontend/ui-customer/src/Services/API.purs [hunk 3]
```diff
@@ -1396,13 +1396,26 @@ instance encodeGetEmergContactsResp  :: Encode GetEmergContactsResp where encode
...
 data GetDriverLocationReq = GetDriverLocationReq String
 
-newtype GetDriverLocationResp = GetDriverLocationResp LatLong
+newtype Location = Location 
+  {
+    lat :: Number,
+    lon :: Number,
+    lastUpdate :: String
+  }
+newtype GetDriverLocationResp = GetDriverLocationResp Location
```
Comment (shanm-16) [inline Frontend/ui-customer/src/Services/API.purs:1405]: there is space between 2 newtypes

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java (hunk 2 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java:129]: yes .

**Comment by witcher-shailesh** [comment]
Not required as of now

**Pull request closed without merging**